camayoc/tests/qpc/cli/conftest.py

```diff
@@ -76,7 +77,13 @@ def cleanup_server():
     We must delete all objects on the server in the correct order, first scans,
     then sources, then credentials.
     """
+    error_finder = re.compile('internal server error')
+    errors = []
     for command in ('scan', 'source', 'cred'):
-        clear = pexpect.spawn('qpc {} clear --all'.format(command))
-        assert clear.expect(pexpect.EOF) == 0
-        clear.close()
+        clear_output = pexpect.run(
+                                    'qpc {} clear --all'.format(command),
+                                    encoding='utf8',
+                                    withexitstatus=True
+                                    )
+        errors.extend(error_finder.findall(str(clear_output)))
```

Comment:
I am curious why casting to `str` is needed for the output.

Reply:
If you set `withexitstatus` to True it will return a tuple instead with `output, exitcode`. You can either call like `output, _ = pexpect.run(..., withexitstatus=True)` or just expect the output and removing the `withexitstatus` argument.

Worth checking the docs for more information about the `run` function  https://pexpect.readthedocs.io/en/stable/api/pexpect.html#pexpect.run

Reply:
rtfd :grin: OK will do

Reply:
Think I addressed you concerns, and now raising all output in assertion error so test result reader has information about which calls to clear failed.